02-documentation/developer-guides/building-a-nextjs-blog-with-squidex.md

```diff
@@ -0,0 +1,588 @@
+
+To make this data available to the `Home` page component of `index.js`, pass posts as a prop to `Home` in index.js:
+
+```js
+export default function Home({ posts }) {
+...
+}
+```
+
+Although the title and slugs of your blog posts are now available to the browser, you cannot see them yet. You will now create a `BlogItem` component that will be used to display and link to all your posts.
+
+Create a `blogItem.js` file in the `squidex-blog/components` folder with the following contents:
+
+```jsx
+import Link from "next/link";
+
+export default function BlogItem({ title, slug }) {
+  return (
+    <div className="blog-item">
+      <Link href={slug}>
+        <a className="blog-item-title">{title}</a>
+      </Link>
+    </div>
+  );
+}
+```
+
+To use `BlogItem` on your homepage, import it in `pages/index.js`
+
+```jsx
+import BlogItem from "../components/blogItem";
+```
+
+And in the `Home` function in `index.js`, after the `Head` tag, add the following:
+
+```jsx
+<div className="blog-post-list">
+  {posts.map((post) => (
+    <BlogItem
+      title={post.flatData.title}
+      slug={post.flatData.slug}
+      key={post.id}
+    />
+  ))}
+</div>
+```
+
+This loops through your posts and creates a link to each of them.
+
+Your completed `index.js` file will look like this:
+
+```jsx
+import Head from "next/head";
+import fetchAPI from "../lib/squidex";
+import BlogItem from "../components/blogItem";
+
+export default function Home({ posts }) {
+  return (
+    <div>
+      <Head>
+        <title>My blog</title>
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <div className="blog-post-list">
+        {posts.map((post) => (
+          <BlogItem
+            title={post.flatData.title}
+            slug={post.flatData.slug}
+            key={post.id}
+          />
+        ))}
+      </div>
+    </div>
+  );
+}
+
+export async function getStaticProps() {
+  const data = await fetchAPI(`
+  {
+    queryPostsContents {
+      id,
+      flatData {
+        title,
+        slug,
+      }
+    }
+  }
+  `);
+  return {
+    props: {
+      posts: data.queryPostsContents,
+    },
+  };
+}
+```
+
+If you look at your blog homepage in a browser, you will see that the titles of your posts are shown and they link to the page where the article can be read.
+
+Now that you have successfully created links to your posts on the homepage, you will create the pages where your blog posts can be read.
+
+## Step 7: Creating the blog post pages
+
+In the `pages` folder, create a `[slug].js` file. This kind of route in a Next.js app is known as a dynamic route. Learn more about dynamic routes and how they are used from [the Next.js docs](https://nextjs.org/docs/routing/dynamic-routes).
```

Comment:
"This kind of route" <-- makes only sense if you have mentioned the term "route" before.

Reply:
Perhaps it would make sense to mention pages first:

In Next.js a page is a React Component file in the `pages` directory. For example, if you a component at `pages/about.js` it will be accessible at `/about`. The `[slug].js` notation is like a fallback and will Next.js to show our page for all URLs that are not handled by another page and to pass over the URL to a variable called `slug`. So when you visit a URL like `/my-first-blog-post` this page is used and `slug` will have the value `my-first-blog-post`.